### DISCUSSION TITLE
Can I rely on  Zustand slice for useState initial value?

### DISCUSSION THREAD

**Discussion by hexwit** [pmndrs/zustand#1146] [Q&A] [answered]
I have something similar to this: 

```
const categories = useStore(store => store.categories);
const [state, setState] = useState(categories.map(...etc));
```

I want to take a slice as an original data, convert it for local consumption and give it as an initial value for local store.  
I want to give possibility to modify local state in a different way before final result gets back to the store. 

So my question is how safe is this approach? Is it possible that local state get some undefined 
value or stale slice because of async nature or hooks? 

Is there better alternative?

**Comment by dai-shi** [comment, marked as answer]
The store should be initialized at module level, so this approach is safe.

**Reply by hexwit** [reply]
Thanks! Is there any better approach than this? Or it is fine?

**Reply by dai-shi** [reply]
Hm, yeah, if you just want to initialize `useState` with store value, this would work better:
```js
  const [state, setState] = useState(() => useStore.getState().categories.map(...etc));
```

**Reply by hexwit** [reply]
Got it, thank you
